src/custom/hooks/useWalletInfo.ts

```diff
@@ -98,7 +98,11 @@ export function useWalletInfo(): ConnectedWalletInfo {
     if (!chainId || !account || !walletName || !GNOSIS_SAFE_WALLET_NAMES.includes(walletName)) {
       setGnosisSafeInfo(undefined)
     } else {
-      getSafeInfo(chainId, account).then(setGnosisSafeInfo)
+      getSafeInfo(chainId, account)
+        .then(setGnosisSafeInfo)
+        .catch((error) => {
+          console.error('[api/gnosisSafe] Error fetching GnosisSafe info', error)
```

Comment:
Probably a good momento to use the retry function.

The retry would allow you to retry and also return the cancel method, that should be called when in the cleanup of the hook